### PR TITLE
Fixed PyQt5 detection without QT_API env var

### PR DESCRIPTION
**Pull request by PierreRaybaut**
When QT_API environment variable has not been set properly, which is safe to assume when using PyQt5 for the first time, importing the `spyderlib.qt` package will fail. This change allows PyQt5 to be imported and used if QT_API has not been set and if both PyQt4 and PySide packages are not installed.

### CODE REVIEW

File: spyderlib/qt/__init__.py [hunk 1]
```diff
@@ -57,6 +54,17 @@
             API_NAME += (" (API v%d)" % sip.getapi('QString'))
         except AttributeError:
             pass
+        from PyQt4 import uic  # analysis:ignore
```
Comment (ccordoba12) [inline spyderlib/qt/__init__.py]: Why is this necessary now?

File: spyderlib/qt/__init__.py [hunk 2]
```diff
@@ -57,6 +54,17 @@
...
+
+PYQT5 = False
+if API == 'pyqt5':
+    try:
+        from PyQt5.QtCore import PYQT_VERSION_STR as __version__
+        from PyQt5 import uic  # analysis:ignore
```
Comment (ccordoba12) [inline spyderlib/qt/__init__.py]: Also this `import`? Do you need it for other things?

**Comment by ccordoba12** [comment]
Thanks a lot @PierreRaybaut. Everything seems fine except for the two comments I made :-)

File: spyderlib/qt/__init__.py (hunk 1 above)
Comment (PierreRaybaut) [inline spyderlib/qt/__init__.py]: The `spyderlib.qt` package was wrote not only for Spyder but also for other libraries (`guidata`, `winpython` and now `python-qwt`). Adding the line `from PyQt4 import uic` here will simply allow to import the `uic` module with `from spyderlib.qt import uic`, that's all. FYI, the `uic` module is used to run GUI built with the help of QtDesigner (see [this example](https://github.com/PierreRaybaut/guiqwt/blob/master/guiqwt/qtdesigner.py)).

File: spyderlib/qt/__init__.py (hunk 2 above)
Comment (PierreRaybaut) [inline spyderlib/qt/__init__.py]: Same explanation as above.

File: spyderlib/qt/__init__.py (hunk 1 above)
Comment (ccordoba12) [inline spyderlib/qt/__init__.py]: Ok, no problem. I just wanted to know why you decided to add it given that we don't use QtDesigner in Spyder.

By the way, we're planning to move to use [qtpy](https://github.com/spyder-ide/qtpy) (one of the Spyder org projects) as our Qt shim.

I think that would help your projects too, to avoid copy/pasting the Spyder shim around, and to have everything consolidated in one place.

**Comment by ccordoba12** [comment]
@PierreRaybaut, just a simple recommendation for next time: please use your own fork of Spyder to create pull requests, instead of creating branches in the main repo.

The thing is those branches are propagated to all people that pull regularly from Github (like our other core developers :-)

Of course, you're more than welcome to submit directly to `master` if you prefer to do so :-)

**Comment by PierreRaybaut** [comment]
@ccordoba12 Sorry for the inconvenience, I'm such a newbie with `git` and GitHub. I'll remember it for next time. Keep up the good work.

**Comment by ccordoba12** [comment]
No problem :-) Glad to have you back!
